src/dataproxy/worker/shared-worker.ts

```diff
@@ -34,43 +55,45 @@ const dataProxy: DataProxyWorker = {
         version: '1'
       },
       schema,
-      store: true
+      store: true,
+      links: [new PouchLink(pouchLinkOptions)]
     })
     client.instanceOptions = clientData.instanceOptions
     client.capabilities = clientData.capabilities
-    if (!searchIndexes) {
-      const indexes = await initIndexes(client)
-      searchIndexes = indexes
-      updateState()
-    }
+
+    searchEngine = new SearchEngine(client)
+
+    updateState()
   },
+
   search: async (query: string) => {
-    log.debug('Received data for search')
     if (!client) {
       throw new Error(
-        'Client is required to execute a seach, please initialize CozyClient'
+        'Client is required to execute a search, please initialize CozyClient'
       )
     }
-
-    if (!searchIndexes) {
-      return []
+    if (!searchEngine) {
+      throw new Error('SearchEngine is not initialized')
     }
 
-    return search(query, searchIndexes, client)
+    return searchEngine.search(query)
   }
 }
 
 const updateState = (): void => {
   const state = {} as DataProxyWorkerPartialState
 
-  if (client && searchIndexes) {
+  if (client && searchEngine && searchEngine.searchIndexes) {
     state.status = 'Ready'
-    state.indexLength = searchIndexes.map(searchIndex => ({
-      doctype: searchIndex.doctype,
-      // @ts-expect-error index.store is not TS typed
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      count: Object.keys(searchIndex.index.store).length
-    }))
+    state.indexLength = Object.keys(searchEngine.searchIndexes).map(
```

Comment:
nit: here you can use `Object.entries().map(([key, value]) => {...}` instead of `Object.keys`
```js
    state.indexLength = Object.entries(searchEngine.searchIndexes).map(
      ([key, value]) => ({
        doctype: key,
        count: Object.keys(value.index.store).length
      })
```

Reply:
Noted, I'll make the change in another PR